libs/hwdrivers/src/rplidar/src/rplidar_driver.cpp

```diff
@@ -349,10 +570,7 @@ u_result RPlidarDriverSerialImpl::ascendScanData(rplidar_response_measurement_no
     }
 
     // all the data is invalid
-    if (i == count){
-        delete[] tmpbuffer;
-        return RESULT_OPERATION_FAIL;
-    }
+    if (i == count) return RESULT_OPERATION_FAIL;
 
     //Tune tail
     for (i = count - 1; i >= 0; i--) {
```

Comment:
it may be a code warning, but not occure error.   I checked it , if "576" invalid,continue i==0, means the data all invalid, equal to i==count,  it does not occur at "576",end at "573".  if have valid data,  for will break at [568](https://github.com/kintzhao/mrpt/blob/master/libs/hwdrivers/src/rplidar/src/rplidar_driver.cpp#L568)